### PR TITLE
Generate builtin tasks lookup file

### PR DESCRIPTION
**Pull request by jessehouwing**
Needs:

- [ ] Access to an Azure DevOps Org. Currently configured against my personal brazil org (gets the tasks the first) used for other similar projects.
- [ ] An Azure DevOps PAT with Agent Pools (Read) permissions in the org above.

Fixes: #12

### CODE REVIEW

File: .github/workflows/create-renovate-data.yml [hunk 1]
```diff
@@ -27,7 +27,14 @@ jobs:
...
-          .\02-generate-renovate-data.ps1
+          .\02-generate-renovate-data-marketplace.ps1
         shell: pwsh 
+
+      - name: Generate azure-pipelines-builtin-tasks.json
+        run: |
+          .\02-generate-renovate-data-builtin.ps1
+        env:
+          AZURE_DEVOPS_PAT: ${{ secrets.AZURE_DEVOPS_PAT }}
+        shell: pwsh
```
Comment (viceice) [inline .github/workflows/create-renovate-data.yml:40]: please add a trailing newline

**Review by viceice — COMMENTED** [review]
@rarkins @JamieMagee we need to decide which PAT we use for this.

**Comment by jessehouwing** [comment]
@rarkins I moved the org name to an Action variable, that way it's under your control :).

**Comment by jessehouwing** [comment]
The generated file using this approach is much smaller by the way (about 1/4rd of the size). I can explain most of that because by far not all versions in the git repo are actually shipped to clients, but that change might warrant a release note in itself. I haven't fully diffed the 2 files yet to see if I can explain all of the differences.

**Comment by jessehouwing** [comment]
https://github.com/renovatebot/azure-devops-marketplace/issues/12#issuecomment-1551947783

**Review by viceice — DISMISSED** [review]
can we start with moving the existing generator from renovate repo here and then later start implementing that api query directly into the renovate datasource.

**Comment by jessehouwing** [comment]
I could take the existing Json from the current version that ships with renovate, commit it and then add the data from the API call...

**Comment by viceice** [comment]
> I could take the existing Json from the current version that ships with renovate, commit it and then add the data from the API call...

lets do it to get it out of the renovate repo

**Review by viceice — APPROVED** [review]
i can't provide appropriate credentials, so if I merge this, it'll fail?

**Comment by jessehouwing** [comment]
> i can't provide appropriate credentials, so if I merge this, it'll fail?

Probably. I can use mine if needed. The tokens are read-only anyway. I just need a safe way to either set them or to send them your way.

**Comment by jessehouwing** [comment]
What I did for my own purposes is:

 * Create a new free Azure DevOps org and I picked the Brazil region.
 * Created a PAT with `Marketplace (Read)` and `Agent Pool (Read)` permissions with that org as scope.

The organization is completely empty otherwise.

The org name is `jessehouwing-brazil`, if you grant me (temporary) admin permissions, I can set the token and variable.

**Comment by jessehouwing** [comment]
@viceice can I help you setup the right credentials? Or would it be ok to use mine?

File: .github/workflows/create-renovate-data.yml [hunk 2]
```diff
@@ -27,7 +27,15 @@ jobs:
...
-          .\02-generate-renovate-data.ps1
+          .\02-generate-renovate-data-marketplace.ps1
         shell: pwsh 
+
+      - name: Generate azure-pipelines-builtin-tasks.json
+        run: |
+          .\02-generate-renovate-data-builtin.ps1
+        env:
+          AZURE_DEVOPS_PAT: ${{ secrets.AZURE_DEVOPS_PAT }}
+          AZURE_DEVOPS_ORG: ${{ vars.AZURE_DEVOPS_ORG }}
```
Comment (JamieMagee) [inline .github/workflows/create-renovate-data.yml:40]:
```suggestion
          AZURE_DEVOPS_ORG: ${{ secrets.AZURE_DEVOPS_ORG }}
```

**Comment by JamieMagee** [comment]
I generated a PAT in my org and added it to this repo. Both are added as secrets, and the PAT will expire on 9th June 2024.

**Pull request merged**